src/hash.cr

```diff
@@ -912,6 +910,21 @@ class Hash(K, V)
     value
   end
 
+  # Sets the value of *key* to the given *value*.
+  #
+  # If a value already exists for `key`, that (old) value is returned.
+  # Otherwise the given block is invoked and its value is returned.
+  #
+  # ```
+  # h = {} of Int32 => String
+  # h.store(1, "one") { "didn't exist" } # => "didn't exist"
+  # h.store(1, "uno") { "didn't exist" } # => "one"
+  # ```
+  def store(key : K, value : V)
+    updated_entry = upsert(key, value)
+    updated_entry ? updated_entry.value : yield
```

Comment:
`store` might perhaps be considered a symmetric operation to `fetch`. `fetch` yields the key to its block. Should `store` also yield something?

Reply:
Great idea! I just did that, thanks. I forgot `fetch` yielded the key.